qa/tasks/ses_qa.py

```diff
@@ -142,6 +142,40 @@ def begin(self):
                 raise ConfigError(self.err_prefix + "No such method ->{}<-"
                                   .format(method_spec))
 
+    def drive_replace_initiate(self, **kwargs):
+        """
+        Initiate Deepsea drive replacement
+
+        Assumes there is 1 drive not being deployed (1node5disks - with DriveGroup `limit: 4`)
+
+        In order to "hide" an existing disk from the ceph.c_v in teuthology
+        the disk is formatted and mounted.
+        """
+        # :TODO: make this random and find correct disk device
+        osd_id = "2"
+        cmd = """sudo ceph osd tree -f json-pretty |
+                 jq '[.nodes[] | select(.type == \"osd\")] | length'"""
+        total_osds = self.master_remote.sh(cmd)
+        assert int(total_osds) == 4
+        self.master_remote.sh("sudo ceph osd tree --format json > before.json")
+        self.master_remote.sh("sudo salt-run osd.replace {} 2>/dev/null".format(osd_id))
+        self.master_remote.sh("sudo mkfs.ext4 -F /dev/vdd; sudo mount /dev/vdd /mnt")
+        self.master_remote.sh("sudo salt-run disks.c_v_commands 2>/dev/null")
+        # Output is like: ceph-volume lvm batch --no-auto /dev/vdf --yes --osd-ids 2
+
+    def drive_replace_check(self, **kwargs):
+        """
+        Deepsea drive replacement after check
+
+        Replaced osd_id should be back in the osd tree once stage.3 is ran
+        """
+        cmd = """sudo ceph osd tree -f json-pretty |
+                 jq '[.nodes[] | select(.type == \"osd\")] | length'"""
```

Comment:
you could move this `cmd = ...` up to the top of the file, and change `cmd` to `number_of_osds_in_cluster` (or `total_osds`?)

(As it is, this assignment is duplicated in two places, which means if it later gets changed the person changing it might not remember to change it in both places...)